pkg/daemon/update.go

```diff
@@ -1766,11 +1745,6 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 
 // updateOS updates the system OS to the one specified in newConfig
 func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig, osImageContentDir string) error {
-	if !dn.os.IsCoreOSVariant() {
```

Comment:
So this does introduce a difference in behaviour I think. Before we just silently accepted the os as "updated" always on RHEL nodes (not an error if you try to "update OS"), but by removing this and passing it off to the notCOS interface you error instead.